fieldpath/element.go

```diff
@@ -35,7 +35,7 @@ type PathElement struct {
 	// Key selects the list element which has fields matching those given.
 	// The containing object must be an associative list with map typed
 	// elements.
-	Key *value.Map
+	Key *value.FieldList
```

Comment:
Now that it's not a map, I *think* we can just make this a FieldList? A zero-length field list is an invalid key, right? A slice is already a pointer, no need to double-point?

Reply:
Antoine mentioned IRL that the size of this object might be a concern, so let's merge this and we can always change this later.